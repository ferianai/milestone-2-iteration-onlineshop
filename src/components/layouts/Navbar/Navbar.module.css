.navbar{
    color: var(--white);
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 50px;
    gap: 10px;
    padding: 0 5%;
  }
  
  .navbar ul{
    display: flex;
    flex-direction: row;
    gap: 10px;
    list-style-type: none;
    font-size: 20px;
  }