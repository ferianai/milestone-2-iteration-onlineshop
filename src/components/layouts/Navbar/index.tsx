import Link from 'next/link';
import style from './Navbar.module.css';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

const Navbar: React.FC = () => {
    const router = useRouter();
    const [email, setEmail] = useState<string | null>(null);
    const [name, setName] = useState<string | null>(null);

    useEffect(() => {
        // Get user data from localStorage when the component mounts
        setEmail(localStorage.getItem('user_email'));
        setName(localStorage.getItem('user_name'));
    }, []);

    const handleLogout = () => {
        // Clear localStorage items
        localStorage.removeItem('access_token');
        localStorage.removeItem('refresh_token');
        localStorage.removeItem('user_email');
        localStorage.removeItem('user_name');
    
        // Redirect to the login page after logging out
        router.push('/auth/login');
    };

    return (
        <nav className={`${style.navbar} bg-blue-500`}>
            <ul>
                <li>
                    <Link href={"/"}>ShopSmart</Link>
                </li>
                <li>
                    <Link href={"/about"}>About</Link>
                </li>
                <li>
                    <Link href={"/product"}>Product</Link>
                </li>
            </ul>
            <ul>
                <li>
                    <Link href={"/cart"}>
                    <span>
                        <i className="fas fa-shopping-cart"></i> {/* FontAwesome cart icon */}
                    </span>
                    </Link>
                </li>
                {name && (
                <li>
                    <span className='font-bold'>{name}</span>
                </li>
                )}
                <li>
                {email ? (
                    <button onClick={handleLogout} className="font-semibold">Logout</button>
                ) : (
                    <Link href="/auth/login" className="font-semibold">Login <i className="fa-solid fa-arrow-right"></i></Link>
                )}
                </li>

            </ul>
        </nav>
    )
}

export default Navbar;