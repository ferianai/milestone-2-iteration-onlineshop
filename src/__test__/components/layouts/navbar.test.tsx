import { render } from '@testing-library/react';
import Navbar from '@/components/layouts/Navbar';

// Mock the useRouter hook
jest.mock('next/router', () => ({
    useRouter: jest.fn().mockReturnValue({
      pathname: '/',  
      query: {},
      push: jest.fn(),
      replace: jest.fn(),
    }),
  }));

test('navbar renders with ShopSmart text', () => {
  const footerText = render(<Navbar />);
  expect(footerText).toMatchSnapshot();
});
