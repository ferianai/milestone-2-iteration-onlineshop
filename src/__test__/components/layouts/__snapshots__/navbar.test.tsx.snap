// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders with ShopSmart text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar bg-blue-500"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              ShopSmart
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/product"
            >
              Product
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              href="/cart"
            >
              <span>
                <i
                  class="fas fa-shopping-cart"
                />
                 
              </span>
            </a>
          </li>
          <li>
            <a
              class="font-semibold"
              href="/auth/login"
            >
              Login 
              <i
                class="fa-solid fa-arrow-right"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar bg-blue-500"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            ShopSmart
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/product"
          >
            Product
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            href="/cart"
          >
            <span>
              <i
                class="fas fa-shopping-cart"
              />
               
            </span>
          </a>
        </li>
        <li>
          <a
            class="font-semibold"
            href="/auth/login"
          >
            Login 
            <i
              class="fa-solid fa-arrow-right"
            />
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
